test/controllers/corrections/CorrectionReturnSinglePeriodControllerSpec.scala

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.corrections
+
+import base.SpecBase
+import controllers.routes
+import forms.corrections.CorrectionReturnSinglePeriodFormProvider
+import models.etmp.{EtmpObligationDetails, EtmpObligationsFulfilmentStatus}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchersSugar.eqTo
+import org.mockito.Mockito.{times, verify, when}
+import org.scalatestplus.mockito.MockitoSugar
+import pages.corrections.CorrectionReturnSinglePeriodPage
+import play.api.data.Form
+import play.api.inject.bind
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import repositories.SessionRepository
+import services.ObligationsService
+import utils.ConvertPeriodKey
+import utils.FutureSyntax.FutureOps
+import views.html.corrections.CorrectionReturnSinglePeriodView
+
+import scala.concurrent.Future
+
+class CorrectionReturnSinglePeriodControllerSpec extends SpecBase with MockitoSugar {
+
+
+  private val periodKeys = Seq("23AK")
+
+  private val periodYear: Seq[Int] = periodKeys.map { periodYear =>
+    s"20${periodYear.substring(0, 2)}".toInt
+  }
+
+  private val monthName: Seq[String] = periodKeys.map(ConvertPeriodKey.monthNameFromEtmpPeriodKey)
+
+  private val monthAndYear = s"${monthName.mkString(", ")} ${periodYear.mkString(", ")}"
```

Comment:
Could we simplify this part by not using `Seq` for `periodKeys` and have the following instead:
```
  private val periodKey = "23AK"

  private val periodYear: Int = s"20${periodKey.substring(0, 2)}".toInt

  private val monthName: String = ConvertPeriodKey.monthNameFromEtmpPeriodKey(periodKey)

  private val monthAndYear = s"$monthName $periodYear"
  ```